### ISSUE TITLE
support addTypename as an option.

### ISSUE THREAD

**Issue by kgdev** [apollographql/graphql-tag#71]

**Comment by jnwng** [comment]
@kgdev just to qualify, can you describe some intended behavior? do you want to be able to automatically add `__typename` to every selection set when using `graphql-tag`?

**Comment by kgdev** [comment]
yes. at least with a option like import 'some.graphql?typename=true'. And we can enable this by default.

**Comment by kasbah** [comment]
Taking a queue from existing options I guess it should be implemented to be used like this?

```js
import gql, { enableTypenameInsertion } from 'graphql-tag'

enableTypenameInsertion()
```

**Comment by dmhalejr** [comment]
👋 I'd like to take this on. My approach was just to use the [apollo-utils library](https://github.com/apollographql/apollo-client/blob/master/packages/apollo-utilities/src/transform.ts#L210) to do the heavy lifting and add typename appropriately
